### DISCUSSION TITLE
Populate Many To Many Table's Foreign Key Relationships

### DISCUSSION THREAD

**Discussion by lanbau** [supabase/supabase#6894] [Questions] [answered]
Hi guys, database noob here.

i managed to get a junction table for https://github.com/supabase/supabase/discussions/710#discussioncomment-362651

And i'm able to query for 

Table 1 - users
id, name

Table 2 - stores
id, name

Junction Table - user_store
id, user_id, store_id

`
const { data, error, status } = await supabase.from("user_store")
        .select(
          `store_id,
            stores:store_id ( name )
          `
        )
        .eq("user_id", user.id);
`

This gives me back the store details that belong to a user which is perfect. 

However, i would like to store more data like chats and it could be a one to many relationship

chats table
id (use user_store primary key?), message, user_id, store_id

Q1. Does it make sense to use the user_store primary key as the primary key id of chats table?

Q2. How can i query for my store's chats using supabase?

**Comment by steve-chavez** [comment, marked as answer]
Hey there,

> Junction Table - user_store
> id, user_id, store_id

You could also make a composite primary key with user_id, store_id. It would save you one column.

> Q1. Does it make sense to use the user_store primary key as the primary key id of chats table?

If the relationship between `chats` and `user_store` is a one-to-one, then yes.

> Q2. How can i query for my store's chats using supabase?

Following your example, it would be:

```js
const { data, error, status } = 
await supabase.from("user_store") .select(`store_id,
  stores(name)
  chats(message)
`) .eq("user_id", user.id);
```

**Reply by lanbau** [reply]
thanks @steve-chavez for the composite primary key.. what would the value be?

if user_id is 1, store_id is 1, is the composite key 11?

**Reply by steve-chavez** [reply]
Yes, that part is optional though, you can have a surrogate key for the junction table.
